documentation/guide-openapi.asciidoc

```diff
@@ -0,0 +1,80 @@
+:toc: macro
+toc::[]
+
+= OpenAPI
+
+The link:https://spec.openapis.org/oas/latest.html[OpenAPI Specification (OAS)] defines a standard for describing RESTful web services in a machine- and human-readable format. OpenAPI allows REST APIs to be defined in a uniform manner.
+Technically, an OpenAPI document is written in YAML or JSON format. The specification defines the structure of a REST API by describing attributes such as path information, response codes, and return types. Some examples can be found link:https://github.com/OAI/OpenAPI-Specification/tree/main/examples/v3.0[here].
+
+OpenAPI is often used in combination with link:https://swagger.io/[Swagger]. Swagger is a set of tools build around OpenAPI, that help developers to design and document their REST APIs.
+The most common tool is the link:https://swagger.io/tools/swagger-ui/[Swagger UI], which uses the OpenAPI specification to create a graphical interface of the REST API that you can also interact with. Check out the link:https://editor.swagger.io/[Swagger online editor] to get a feeling for it.
+
+== OpenAPI generation
+
+There are several extensions you can use in your project to automatically generate the OpenAPI specifications and Swagger UI from your REST API (code-first approach). devon4j recommends the following two extensions/plugins to use:
+
+* Smallrye OpenAPI extension (for Quarkus)
```

Comment:
Do we?
I thought we recommend ServiceDocGen also for Quarkus if you don't want to re-specify all rest service information again as annotations. ServiceDocGen should be also very interesting for the Quarkus stack in this case when implementing JAX-RS standard.

Reply:
Would be great to say, Quarkus recommends Smallrye whereas devonfw recommends strongly to have a look at ServiceDocGen if you want to reduce effort on REST service documentation by omitting additional documentation annotations needed for smallrye. Nonetheless, in smallrye, you can document services in much more detail if needed.

Reply:
Did some rework on it